porthole/reports.py

```diff
@@ -300,7 +302,7 @@ def execute(self):
             if self.check_whether_to_publish():
                 self.publish()
             if self.db_logger is not None:
-                self.db_logger.finalize_record()
+                self.db_logger.finalize_record(self.all_recipients)
```

Comment:
As written, I believe this will log recipients even if `check_whether_to_publish` returns `False`. A condition to pass in the recipients array only when the report has been published would resolve this.